### PR TITLE
update HLL++ to use sparse mode

### PR DESCRIPTION
**Pull request by jgoodknight**
HLL++ doesn't do well for small cardinalities so we are updating our implementation to use a version of sparse mode from the HLL++ paper, that doesn't care so much about memory space but is much simpler to implement.

Note this is not tested.

### CODE REVIEW

**Review by kungfucraig — CHANGES_REQUESTED** [review]
Please add tests, especially for the merge.

**Comment by huangxichen1** [comment]
> Please add tests, especially for the merge.

Added tests.

> Please add tests, especially for the merge.

I added the tests. PTAL. Thanks!

File: src/estimators/tests/hyper_log_log_test.py [hunk 1]
```diff
@@ -155,6 +157,65 @@ def test_insert_large(self):
...
+
+    merged_hll = hll1.merge(hll2)
+    self.assertTrue(merged_hll.sparse_mode,
+                    'Merged sketch should be in sparse mode.')
+
+    hll1.add(1000)
+    merged_hll = hll1.merge(hll2)
+    self.assertFalse(merged_hll.sparse_mode,
+                     'Merged sketch should not be in sparse mode.')
+
```
Comment (kungfucraig) [inline src/estimators/tests/hyper_log_log_test.py:187]: Can you test the cardinality of the merged hll is as expected?

**Review by kungfucraig — APPROVED** [review]

File: src/estimators/tests/hyper_log_log_test.py (hunk 1 above)
Comment (huangxichen1) [reply src/estimators/tests/hyper_log_log_test.py:196]: I added tests for the estimated cardinality.

**Pull request merged**